src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const backupObj = Object.assign({}, state);
+
+  actions.forEach(action => {
+    switch (action.type) {
+      case 'addProperties':
+        for (const element of Object.entries(action.extraData)) {
+          backupObj[element[0]] = element[1];
+        }
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          if (key in backupObj) {
+            delete backupObj[key];
+          }
+        }
+        break;
+
+      case 'clear':
+        for (const key in backupObj) {
+          delete backupObj[key];
+        }
```

Comment:
why not just set `{}` instead of deleting props?

Reply:
Very true